scss/_buttons.scss

```diff
@@ -43,38 +57,82 @@
   &.disabled,
   fieldset:disabled & { // stylelint-disable-line selector-no-qualifying-type
     pointer-events: none;
+    // Remove CSS gradients if they're enabled
+    background-image: if($enable-gradients, none, null);
     opacity: $btn-disabled-opacity;
     @include box-shadow(none);
   }
 }
 
+.btn-outline {
+  color: var(--btn-accent-color);
+  background-color: transparent;
+  border-color: currentColor;
+
+  &:hover {
+    background-color: var(--btn-accent-color);
+    border-color: var(--btn-accent-color);
+  }
+
+  &:focus,
+  &.focus {
+    color: var(--btn-accent-contrast-color);
+    background-color: var(--btn-accent-color);
+    border-color: var(--btn-accent-color);
+  }
+
+  &.active,
+  &:active {
+    color: var(--btn-accent-contrast-color);
+    background-color: var(--btn-accent-color);
+    border-color: var(--btn-accent-color);
+  }
+}
 
 //
 // Alternate buttons
 //
 
 @each $color, $value in $theme-colors {
   .btn-#{$color} {
-    @include button-variant($value, $value);
-  }
-}
+    $contrast-color: color-contrast($value);
 
-@each $color, $value in $theme-colors {
-  .btn-outline-#{$color} {
-    @include button-outline-variant($value);
+    --btn-accent-color: #{$value};
```

Comment:
I'm a bit confused by the fact that the variables are set here. The nice thing about CSS variables are that they cascade through the DOM hierarchy, so that devs can set these e.g. at the `:root` or some other ascendant of a button, and every button (and other element that was configured in the ancestor) below that ancestor will use the same theme - without having to use any descendant selectors for `.btn` etc.

But if the variable is set here on the `.btn` element, then it seems that wouldn't work, because it would always override what was set in an ancestor. So to make it work, define a theme for all descendant Bootstrap components at an ancestor element, it seems we'd have to use specific descendant selectors like `.btn` again to override the variables. That kind of defeats the purpose of the variables mostly.

How I imagine it, components like `.btn` would only _read_ variables, and they are set e.g. on `:root` or any ancestor by devs (and Bootstrap can provide defaults on those and integrate them there with the Sass variables).

Reply:
I have the same expression, looks like CSS custom properties here used to refactor code in a cool way, not to increase customization capabilities and flexibility...